MoviesMVP/MoviesMVPTests/NetworkServiceTests.swift

```diff
@@ -0,0 +1,123 @@
+// NetworkServiceTests.swift
+// Copyright © RoadMap. All rights reserved.
+
+import XCTest
+
+@testable import MoviesMVP
+
+/// Network service tests
+final class NetworkServiceTests: XCTestCase {
+    // MARK: - Public properties
+  
+    var networkService: NetworkServiceProtocol?
+    var movies: [Movie]?
+    var movieDetail: MovieDetail?
+    var cast: [Cast]?
+
+   // MARK: - Public methods
+  
+    override func setUp() {
+        networkService = MockNetworkService()
+    }
+
+    override func tearDown() {
+        networkService = nil
+    }
+
+    func testSuccessMovies() {
+        movies = [Movie]()
+        networkService = MockNetworkService(movies: movies)
+        var catchMovies: [Movie]?
+
+        networkService?.fetchMovies(categoryOfMovies: "popular", page: 1) { result in
+            switch result {
+            case let .success(movies):
+                catchMovies = movies
+            case .failure:
+                break
+            }
+        }
+        XCTAssertEqual(catchMovies?.count, 1)
+        movies = nil
+    }
+
+    func testFailureMovies() {
+        networkService = MockNetworkService(movies: movies)
+        var catchError: Error?
+        networkService?.fetchMovies(categoryOfMovies: "popular", page: 1) { result in
+            switch result {
+            case .success:
+                break
+            case let .failure(error):
+                catchError = error
+            }
+        }
+        XCTAssertNotNil(catchError)
+    }
+
+    func testSuccessMovieDetail() {
+        movieDetail = MovieDetail()
+        networkService = MockNetworkService(movieDetail: movieDetail)
+        let movieID = "550"
+        var catchMovieDetail: MovieDetail?
+
+        networkService?.fetchDetails(id: movieID) { result in
+            switch result {
+            case let .success(movie):
+                catchMovieDetail = movie
+            case .failure:
+                break
+            }
+        }
+        XCTAssertEqual(catchMovieDetail?.id, Int(movieID))
+        movieDetail = nil
+    }
+
+    func testFailureMovieDetail() {
+        networkService = MockNetworkService(movieDetail: movieDetail)
+        let movieID = "550"
+        var catchError: Error?
+        networkService?.fetchDetails(id: movieID) { result in
+            switch result {
+            case .success:
+                break
+            case let .failure(error):
+                catchError = error
+            }
+        }
+        XCTAssertNotNil(catchError)
+    }
+
+    func testSuccessCast() {
+        cast = [Cast]()
+        networkService = MockNetworkService(cast: cast)
+        var catchCast: [Cast]?
+        let movieID = "411"
+
+        networkService?.fetchCast(id: movieID) { result in
+            switch result {
+            case let .success(cast):
+                catchCast = cast
+            case .failure:
+                break
+            }
+        }
+        XCTAssertEqual(catchCast?.count, 6)
```

Comment:
```suggestion
        XCTAssertEqual(catchCast?.count, 6)
```
это число я бы тоже заменил константой